service/src/main/java/io/camunda/service/DocumentServices.java

```diff
@@ -83,49 +81,47 @@ public CompletableFuture<DocumentReferenceResponse> createDocument(
     final List<Either<DocumentErrorResponse, DocumentReferenceResponse>> results =
         new ArrayList<>();
 
-    final List<CompletableFuture<Void>> futures =
+    final List<CompletableFuture<Boolean>> futures =
         requests.stream()
             .map(
                 request -> {
                   final var storeRequest =
                       new DocumentCreationRequest(
                           request.documentId, request.contentInputStream, request.metadata);
-                  return getDocumentStore(request.storeId)
-                      .thenCompose(
-                          storeRecord ->
-                              storeRecord
-                                  .instance()
-                                  .createDocument(storeRequest)
-                                  .thenApply(
-                                      result ->
-                                          transformResponse(
-                                              request, result, storeRecord.storeId())))
-                      .thenAccept(results::add);
+                  final DocumentStoreRecord documentStore = getDocumentStore(request.storeId);
```

Comment:
💭 Just commenting for clarity: We don't chain the `getDocumentStore()` method, but use a `documentStore` variable on [line 99](https://github.com/camunda/camunda/pull/35359/files#diff-68f5d6c7fdd3689b52b81f89f94193806fa90472f388700fbffb122375b931dfR99) since the `storeId` might be different from the `request.storeId` (ex. if the `request.storeId` is `null`).